test/e2e/imagelist_alias/eraser_test.go

```diff
@@ -0,0 +1,179 @@
+//go:build e2e
+// +build e2e
+
+package e2e
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	eraserv1alpha1 "github.com/Azure/eraser/api/v1alpha1"
+	"github.com/Azure/eraser/test/e2e/util"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"sigs.k8s.io/e2e-framework/klient/wait"
+	"sigs.k8s.io/e2e-framework/klient/wait/conditions"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+)
+
+func TestRemoveImagesFromAllNodes(t *testing.T) {
+	const (
+		nginx         = "nginx"
+		nginxLatest   = "docker.io/library/nginx:latest"
+		nginxAliasOne = "docker.io/library/nginx:one"
+		nginxAliasTwo = "docker.io/library/nginx:two"
+		redis         = "redis"
+		caddy         = "caddy"
+
+		prune               = "imagelist"
+		skippedNodeName     = "eraser-e2e-test-worker"
+		skippedNodeSelector = "kubernetes.io/hostname=eraser-e2e-test-worker"
+		skipLabelKey        = "eraser.sh/cleanup.skip"
+		skipLabelValue      = "true"
+	)
+
+	aliasFix := features.New("Specifying an image alias in the image list will delete the underlying image").
+		// Deploy 3 deployments with different images
+		// We'll shutdown two of them, run eraser with `*`, then check that the images for the removed deployments are removed from the cluster.
+		Setup(func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			// Ensure that both nginx:one and nginx:two are tags for the same image digest
+			_, err := util.DockerPullImage(nginxLatest)
+			if err != nil {
+				t.Error("failed to pull nginx image", err)
+			}
+
+			// Create the alias nginx:one
+			_, err = util.DockerTagImage(nginxLatest, nginxAliasOne)
+			if err != nil {
+				t.Error("failed to tag nginx image", err)
+			}
+
+			// Create the alias nginx:two
+			_, err = util.DockerTagImage(nginxLatest, nginxAliasTwo)
+			if err != nil {
+				t.Error("failed to tag nginx image", err)
+			}
+
+			// Load the images into the cluster
+			_, err = util.KindLoadImage(util.KindClusterName, nginxAliasOne)
+			if err != nil {
+				t.Error("failed to load kind image", err)
+			}
+
+			_, err = util.KindLoadImage(util.KindClusterName, nginxAliasTwo)
```

Comment:
can we pass multiple images to `KindLoadImage`? it's not in the PR scope, we can create an issue for this too